### PR TITLE
Update transactor for M2 mainnet contracts

### PR DESCRIPTION
**Pull request by mooselumph**
## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: core/attestation.go [hunk 1]
```diff
@@ -16,6 +17,21 @@ type G1Point struct {
 	*bn254.G1Affine
 }
 
+func newFpElement(x *big.Int) fp.Element {
+	var p fp.Element
+	p.SetBigInt(x)
+	return p
+}
+
+func NewG1Point(x, y *big.Int) *G1Point {
```
Comment (jianoaix) [inline core/attestation.go:26]: Not used here, is it for the other PR? If so maybe move it over there.

**Review by jianoaix — APPROVED** [review]

File: core/eth/tx.go [hunk 2]
```diff
@@ -169,22 +98,97 @@ func (t *Transactor) GetRegisteredQuorumIdsForOperator(ctx context.Context, oper
...
+	ctx context.Context,
+	keypair *core.KeyPair,
+	operatorEcdsaPrivateKey *ecdsa.PrivateKey,
+	operatorToAvsRegistrationSigSalt [32]byte,
+	operatorToAvsRegistrationSigExpiry *big.Int,
+) (*regcoordinator.IBLSApkRegistryPubkeyRegistrationParams, *regcoordinator.ISignatureUtilsSignatureWithSaltAndExpiry, error) {
+
+	operatorAddress := t.EthClient.GetAccountAddress()
+
+	msgToSignG1_, err := t.Bindings.RegistryCoordinator.PubkeyRegistrationMessageHash(&bind.CallOpts{}, operatorAddress)
```
Comment (ian-shim) [inline core/eth/tx.go:111]: reuse `ctx`

File: core/eth/tx.go [hunk 3]
```diff
@@ -169,22 +98,97 @@ func (t *Transactor) GetRegisteredQuorumIdsForOperator(ctx context.Context, oper
...
+	}
+
+	params := regcoordinator.IBLSApkRegistryPubkeyRegistrationParams{
+		PubkeyRegistrationSignature: signedMessageHashParam,
+		PubkeyG1:                    g1Point,
+		PubkeyG2:                    g2Point,
+	}
+
+	// params to register operator in delegation manager's operator-avs mapping
+	msgToSign, err := t.Bindings.DelegationManager.CalculateOperatorAVSRegistrationDigestHash(
```
Comment (ian-shim) [inline core/eth/tx.go:142]: reuse `ctx`

File: core/eth/tx.go [hunk 4]
```diff
@@ -236,13 +245,18 @@ func (t *Transactor) RegisterOperatorWithChurn(ctx context.Context, pubkeyG1 *co
 		t.Logger.Error("Failed to generate transact opts", "err", err)
 		return err
 	}
-	tx, err := t.Bindings.BLSRegCoordWithIndices.RegisterOperatorWithCoordinator(
+
+	// TODO(mooselumph): Fill out these
+	kickParams := []regcoordinator.IRegistryCoordinatorOperatorKickParam{}
```
Comment (ian-shim) [inline core/eth/tx.go:250]: is this to be done in following PR? can we create a ticket to track it

File: core/eth/tx.go [hunk 5]
```diff
@@ -279,21 +293,20 @@ func (t *Transactor) DeregisterOperator(ctx context.Context, pubkeyG1 *core.G1Po
 
 	quorumNumbers := bitmapToBytesArray(quorumBitmap)
 
-	pubkey := pubKeyG1ToBN254G1Point(pubkeyG1)
-	g1Point := regcoordinator.BN254G1Point{
-		X: pubkey.X,
-		Y: pubkey.Y,
-	}
+	// pubkey := pubKeyG1ToBN254G1Point(pubkeyG1)
```
Comment (ian-shim) [inline core/eth/tx.go:296]: remove?

File: core/eth/tx.go [hunk 6]
```diff
@@ -567,20 +580,20 @@ func (t *Transactor) CalculateOperatorChurnApprovalDigestHash(
...
 
-		opKickParams[i] = regcoordinator.IBLSRegistryCoordinatorWithIndicesOperatorKickParam{
+		opKickParams[i] = regcoordinator.IRegistryCoordinatorOperatorKickParam{
 			QuorumNumber: operatorsToChurn[i].QuorumId,
 			Operator:     operatorsToChurn[i].Operator,
-			Pubkey: regcoordinator.BN254G1Point{
-				X: pubkey.X.BigInt(new(big.Int)),
-				Y: pubkey.Y.BigInt(new(big.Int)),
-			},
+			// Pubkey: regcoordinator.BN254G1Point{
```
Comment (ian-shim) [inline core/eth/tx.go:590]: remove?

File: core/eth/tx.go [hunk 7]
```diff
@@ -589,51 +602,67 @@ func (t *Transactor) GetCurrentBlockNumber(ctx context.Context) (uint32, error)
 	return t.EthClient.GetCurrentBlockNumber(ctx)
 }
 
-func (t *Transactor) GetQuorumCount(ctx context.Context, blockNumber uint32) (uint16, error) {
-	return t.Bindings.StakeRegistry.QuorumCount(&bind.CallOpts{
+func (t *Transactor) GetQuorumCount(ctx context.Context, blockNumber uint32) (uint8, error) {
+	//TODO(mooselumph): What is the correct function to use here?
```
Comment (ian-shim) [inline core/eth/tx.go:606]: @0x0aa0 can you chime in?

**Review by ian-shim — COMMENTED** [review]
lgtm, I'm getting this is to be merged after https://github.com/Layr-Labs/eigenda/pull/193?

File: core/attestation.go (hunk 1 above)
Comment (mooselumph) [reply core/attestation.go:26]: Used here: https://github.com/mooselumph/eigenda/blob/accfb672699d22282dc1a1ee8d32e4106a519bd3/core/eth/tx.go#L116

File: core/eth/tx.go (hunk 4 above)
Comment (mooselumph) [reply core/eth/tx.go:250]: Adding a comment to the ticket for the churner updates: https://linear.app/eigenlabs/issue/EGDA-194/churner-updates-for-m2-mainnet-contracts#comment-66163e52

**Review by ian-shim — APPROVED** [review]

**Pull request merged**